time_capture/scripts/attendance.py

```diff
@@ -0,0 +1,53 @@
+import frappe
+from frappe import _
+
+from time_capture.time_capture.doctype.time_capture.time_capture import _create_time_capture
+
+
+def before_insert(doc, event):
+	check_is_compensatory_leave(doc)
+
+
+def on_submit(doc, event):
+	delete_time_capture(doc)
```

Comment:
`on_submit` -> `delete_time_capture` sounds misleading. Better add the guard clause before the function call, not in the function.

```suggestion
def on_submit(doc, event):
	if doc.leave_type and doc.attendance_date <= frappe.utils.nowdate():
		delete_time_capture(doc)
```